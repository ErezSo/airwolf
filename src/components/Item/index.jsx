import React, { Component } from "react";
import { ItemBox, OrderButton } from "../";

const styles = {
  container: {
    height: 300
  },
  itemWrapper: {
    minWidth: 200,
    minHeight: 200,
    boxShadow: "0 0 10px 10px #f5f6f6, 0 0 20px 20px #f5f6f6",
    margin: "10px 10px 20px",
    position: "relative"
  },
  textWrapper: {
    display: "flex",
    justifyContent: "center",
    alignItems: "center"
  },
  itemText: {
    fontSize: 13,
    textAlign: "center",
    width: 164
  }
};

class Item extends Component {
  state = {
    hover: false
  };

  toggleHover = () => {
    this.setState(({ hover }) => ({ hover: !hover }));
  };

  render() {
    const { item, idx, reserveItem } = this.props;
    const { itemWrapper, container, textWrapper, itemText } = styles;
    let buttonContainerStyle, itemOpacity;
    if (this.state.hover) {
      buttonContainerStyle = { display: "block" };
      itemOpacity = { opacity: "0.2" };
    } else {
      buttonContainerStyle = { display: "none" };
      itemOpacity = { opacity: "1" };
    }

    return (
      <div style={container}>
        <div
          style={itemWrapper}
          onMouseEnter={this.toggleHover}
          onMouseLeave={this.toggleHover}
        >
          <OrderButton
            containerStyle={buttonContainerStyle}
            reserveItem={reserveItem}
          />
          <ItemBox item={item} idx={idx} itemOpacity={itemOpacity} />
        </div>
        <div style={textWrapper}>
          <div style={itemText}>
            42mm Silver Aluminum Case with Scuba Blue Woven Nylon
          </div>
        </div>
      </div>
    );
  }
}

export default Item;
