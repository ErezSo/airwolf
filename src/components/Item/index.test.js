import React from "react";
import { shallow } from "enzyme";
import Item from "./";

describe("Item", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Item />);
  });

  it("renders", () => {
    expect(wrapper).toHaveLength(1);
  });

  it("renders correctly", () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
