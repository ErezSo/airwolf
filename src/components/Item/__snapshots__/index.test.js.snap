// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item renders correctly 1`] = `
"<div style={{...}}>
  <div style={{...}} onMouseEnter={[Function]} onMouseLeave={[Function]}>
    <OrderButton containerStyle={{...}} reserveItem={[undefined]} />
    <ItemBox item={[undefined]} idx={[undefined]} itemOpacity={{...}} />
  </div>
  <div style={{...}}>
    <div style={{...}}>
      42mm Silver Aluminum Case with Scuba Blue Woven Nylon
    </div>
  </div>
</div>"
`;
